src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using Effects = Xamarin.CommunityToolkit.iOS.Effects;
+
+[assembly: ExportEffect(typeof(Effects.BackgroundAspectEffectRouter), nameof(BackgroundAspectEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class BackgroundAspectEffectRouter : PlatformEffect
+	{
+		UIImageView? imageView;
+
+		CancellationTokenSource cancellationTokenSource;
+
+		public BackgroundAspectEffectRouter()
+		{
+			cancellationTokenSource = new CancellationTokenSource();
+		}
+
+		protected override void OnAttached()
+			=> Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+
+		protected override void OnDetached()
+			=> ClearBackgroundImage();
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (!args.PropertyName.Equals(BackgroundAspectEffect.AspectProperty.PropertyName))
+				return;
+
+			cancellationTokenSource.Cancel();
+
+			Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
+		}
+
+		async Task ApplyBackgroundImageAsync()
+		{
+			var contentPage = Element as ContentPage;
+
+			if (Element is null || contentPage is null || contentPage.BackgroundImageSource is null)
```

Comment:
we don't check for `null` using pattern matching :(
```suggestion
			if (Element == null || contentPage == null || contentPage.BackgroundImageSource == null)
```

Reply:
No worries, I've recently fallen in love with the pattern matching syntax: `if (something is null)` & `if (something is not null)` makes the code look so nice imo 😁